src/Npgsql/NpgsqlParameter.cs

```diff
@@ -320,12 +316,12 @@ public sealed override DbType DbType
                 Handler = null;
                 if (value == DbType.Object)
                 {
-                    _dbType = null;
+                    _cachedDbType = null;
                     _npgsqlDbType = null;
                 }
                 else
                 {
-                    _dbType = value;
+                    _cachedDbType = value;
                     _npgsqlDbType = GlobalTypeMapper.Instance.ToNpgsqlDbType(value);
```

Comment:
What if I there is a difference between global and connection mappings?

Reply:
Good question. The problem is that the translations between DbType and NpgsqlDbType are expected to work without the parameter being even attached to a command (and so to a connection). The architecture simply doesn't have type mapping configurability in mind really.

However, this is probably only a theoretical problem. If people touch type mapping, it is usually only to change handlers - this doesn't change NpgsqlDbType/DbType translations. Note also that NpgsqlDbType/DbType is only valid for built-in stuff anyway (no plugins).

Reply:
I think if some problems happens, we will fix them later on request.

Reply:
Agreed - it has been this way since forever and nobody has complained...